client/src/rr/dnssec/keypair.rs

```diff
@@ -241,11 +242,12 @@ impl KeyPair {
     /// # Return
     ///
     /// the KEY record data
+    #[allow(deprecated)]
     pub fn to_sig0key(&self, algorithm: Algorithm) -> DnsSecResult<KEY> {
         self.to_public_bytes()
             .map(|bytes| {
                      KEY::new(Default::default(),
-                              Default::default(),
+                              KeyUsage::Zone,
```

Comment:
Well that's only a default value for `dnssec-keygen` (and, according to the manpage, only for `DNSKEY` records).
You can change it with `-n HOST` or `-n USER`.

```
       -n nametype
           Specifies the owner type of the key. The value of nametype must either be ZONE (for a DNSSEC zone key (KEY/DNSKEY)), HOST or ENTITY (for a key
           associated with a host (KEY)), USER (for a key associated with a user(KEY)) or OTHER (DNSKEY). These values are case insensitive. Defaults to
           ZONE for DNSKEY generation.
```